File: zenscript-language-server/src/main/java/raylras/zen/lsp/bracket/BracketHandlerEntry.java
package raylras.zen.lsp.bracket;

import java.util.Collections;
import java.util.List;
import java.util.Map;
import java.util.Optional;

public record BracketHandlerEntry(Map<String, List<String>> properties) {

    public List<String> get(String key) {
        List<String> values = properties.get(key);
        return (values != null) ? values : Collections.emptyList();
    }

    public Optional<String> getFirst(String key) {
        List<String> values = get(key);
        return !values.isEmpty() ? Optional.of(values.get(0)) : Optional.empty();
    }

}

File: zenscript-language-server/src/main/kotlin/raylras/zen/lsp/bracket/BracketHandlerEntry.kt
package raylras.zen.lsp.bracket

data class BracketHandlerEntry(val properties: Map<String, Any>) {
    fun getStringOrNull(key: String): String? {
        return properties[key] as? String
    }

    @Suppress("UNCHECKED_CAST")
    fun getStringListOrNull(key: String): List<String>? {
        return properties[key] as? List<String>
    }
}
